js/start.js

```diff
@@ -312,76 +312,62 @@ function fetchVerifiedMods() {
+  ownFollowingFetch = !ownFollowingFetch || ownFollowingFetch.state() === 'rejected' ?
     app.ownFollowing.fetch() : ownFollowingFetch;
   exchangeRatesFetch = exchangeRatesFetch || fetchExchangeRates();
-  walletBalanceFetch = !walletBalanceFetch || walletBalanceFetchFailed ?
+  walletBalanceFetch = !walletBalanceFetch || walletBalanceFetch.state() === 'rejected' ?
     app.walletBalance.fetch() : walletBalanceFetch;
-  searchProvidersFetch = !searchProvidersFetch || searchProvidersFetchFailed ?
+  searchProvidersFetch = !searchProvidersFetch || searchProvidersFetch.state() === 'rejected' ?
     app.searchProviders.fetch() : searchProvidersFetch;
-  verifiedModsFetch = !verifiedModsFetch || verifiedModsFetchFailed ?
+  verifiedModsFetch = !verifiedModsFetch || verifiedModsFetch.state() === 'rejected' ?
     app.verifiedMods.fetch() : verifiedModsFetch;
 
   $.whenAll(ownFollowingFetch, exchangeRatesFetch, walletBalanceFetch, searchProvidersFetch,
     verifiedModsFetch)
-    .progress((...args) => {
-      const state = args[1];
-
-      if (state !== 'success') {
-        const jqXhr = args[0];
-
-        if (jqXhr === ownFollowingFetch) {
-          ownFollowingFailed = true;
-        } else if (jqXhr === walletBalanceFetch) {
-          walletBalanceFetchFailed = true;
-        } else if (jqXhr === searchProvidersFetch) {
-          searchProvidersFetchFailed = true;
-        } else if (jqXhr === verifiedModsFetch) {
-          verifiedModsFetchFailed = true;
-        }
-      }
-    })
     .done(() => {
       fetchStartupDataDeferred.resolve();
       setInterval(() => fetchVerifiedMods(), 360000);
     })
-    .fail((jqXhr) => {
+    .fail((...args) => {
       const curConn = getCurrentConnection();
 
-      // Check to see if the server is not connected. the search providers and verified moderators
-      // can return an undefined jqXhr if their external endpoints are down while the app itself is
-      // still functional.
-      if ((ownFollowingFailed || walletBalanceFetchFailed) &&
-        (!jqXhr || !curConn || curConn.status !== 'connected')) {
+      if (!curConn || curConn.status !== 'connected') {
         // the connection management modal should be up with relevant info
         console.error('The startup data fetches failed. Looks like the connection to the ' +
           'server was lost.');
         return;
       }
 
+      const ownFollowingFailArgs = args[0];
+      const exchangeRatesFailArgs = args[1];
+      const walletBalanceFailArgs = args[2];
+      const searchProvidersFailArgs = args[3];
+
+      // Find any that failed aside from the exchangeRateFetch. We don't care if the
+      // exchange rate fetch failed, because the exchangeRateSyncer will display a
+      // status message about it and the app will gracefully handle not having exchange
+      // rates.
+      const failed = args.filter(failArgs => failArgs && failArgs !== exchangeRatesFailArgs);
```

Comment:
Is there any chance one of the other calls could fail at the same time as the exchange rates call, and this would discard it because the data is coincidentally the same as the exchange rate fail data?

Could it be done more explicitly with something like args.splice(1,1).filter?

Reply:
> Is there any chance one of the other calls could fail at the same time as the exchange rates call, and this would discard it because the data is coincidentally the same as the exchange rate fail data?

No, it wouldn't be discarded. The only discarded one is the exchange rate one. Any other one would be returned as true by the filter and satisfy the `if (failed.length)` condition below.